controllers/cloudstackmachine_controller.go

```diff
@@ -21,264 +21,223 @@ import (
-			msg := "error patching CloudStackMachine %s/%s"
-			err = errors.Wrapf(err, msg, csMachine.Namespace, csMachine.Name)
-			retErr = multierror.Append(retErr, err)
-		}
-	}()
+// CloudStackMachineReconciler reconciles a CloudStackMachine object
+type CloudStackMachineReconciler struct {
+	csCtrlrUtils.ReconcilerBase
+}
 
-	// Fetch the CAPI Machine.
-	capiMachine, err := util.GetOwnerMachine(ctx, r.Client, csMachine.ObjectMeta)
-	if err != nil {
-		return ctrl.Result{}, err
-	} else if capiMachine == nil {
-		log.Info("Waiting for CAPI cluster controller to set owner reference on CloudStack machine.")
-		return ctrl.Result{RequeueAfter: requeueTimeout}, nil
-	}
+// Initialize a new CloudStackMachine reconciliation runner with concrete types and initialized member fields.
+func NewCSMachineReconciliationRunner() *CloudStackMachineReconciliationRunner {
+	// Set concrete type and init pointers.
+	runner := &CloudStackMachineReconciliationRunner{ReconciliationSubject: &infrav1.CloudStackMachine{}}
+	runner.CAPIMachine = &capiv1.Machine{}
+	runner.Zones = &infrav1.CloudStackZoneList{}
+	runner.IsoNet = &infrav1.CloudStackIsolatedNetwork{}
+	runner.AffinityGroup = &infrav1.CloudStackAffinityGroup{}
+	runner.FailureDomain = &infrav1.CloudStackZone{}
+	// Setup the base runner. Initializes pointers and links reconciliation methods.
+	runner.ReconciliationRunner = csCtrlrUtils.NewRunner(runner, runner.ReconciliationSubject)
+	return runner
+}
 
-	// Fetch the CAPI Cluster.
-	cluster, err := util.GetClusterFromMetadata(ctx, r.Client, capiMachine.ObjectMeta)
-	if err != nil {
-		log.Info("Machine is missing cluster label or cluster does not exist.")
-		return ctrl.Result{RequeueAfter: requeueTimeout}, nil
-	}
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+func (reconciler *CloudStackMachineReconciler) Reconcile(ctx context.Context, req ctrl.Request) (res ctrl.Result, retErr error) {
+	return NewCSMachineReconciliationRunner().
+		UsingBaseReconciler(reconciler.ReconcilerBase).
+		ForRequest(req).
+		WithRequestCtx(ctx).
+		RunBaseReconciliationStages()
+}
 
-	// Check if the machine is paused.
-	if annotations.IsPaused(cluster, csMachine) {
-		log.Info("CloudStackMachine or linked Cluster is paused. Requeuing reconcile.")
-		return reconcile.Result{}, nil
-	}
+func (r *CloudStackMachineReconciliationRunner) Reconcile() (retRes ctrl.Result, reterr error) {
+	return r.RunReconciliationStages(
+		r.GetZones(r.Zones),
+		r.GetParent(r.ReconciliationSubject, r.CAPIMachine),
+		r.RequeueIfCloudStackClusterNotReady,
+		r.ConsiderAffinity,
+		r.SetFailureDomainOnCSMachine,
+		r.GetObjectByName("placeholder", r.IsoNet, func() string { return r.FailureDomain.Spec.Network.Name }),
+		// This can move to IsoNet controller with a nice watch someday.
+		r.GetOrCreateVMInstance,
+		r.RequeueIfInstanceNotRunning,
+		r.AddToLBIfNeeded)
```

Comment:
The comment is in the wrong spot. The add to LB method will hopefully eventually get removed from the machine controller. It is the last remnant of regular CRDs needing to manage Isolated Network concepts.

Reply:
Good future direction. In the mean time, would you move the comment down?

Reply:
Yup!